src/Map.jsx

```diff
@@ -29,9 +32,51 @@ class Map extends React.Component {
       showLocation: false,
       showButton: false,
       bivakzones: props.bivakzones,
+      showModal:false,
+      arrowDirection: false,
+      clicks:0,
+      bivakzone:null
     };
     console.log(this.state.bivakzones);
   }
+  
+  // eslint-disable-next-line no-useless-constructor  
+ 
+ showModalFunc=(modalState)=>{
+   this.setState(
+     {     
+     showModal: modalState
+     }
+   )
+ }
+ 
+ handleOnClick= (e)=>{
+  const prevBivId= this.state.bivakzone;
+   
+   {((e.sourceTarget.feature === prevBivId  )  ? this.setState({showModal:!this.state.showModal, bivakzone: null}) : this.setState({showModal: false,clicks: this.state.clicks + 1,bivakzone: e.sourceTarget.feature}))}
+   
+  //  this.setState({
+  //   bivakzone: e.sourceTarget.feature
+  // })
+   //  this.context= false;
+   console.log((prevBivId === e.sourceTarget.feature)  )
+   console.log(this.state.clicks)
+  //  this.child.showModal()
+  }
+  handlArrowClick=()=>{
```

Comment:
We should refactor this component to define this functions in a seperate component/class to import and use them easily here and whatever we need!